src/value-to-tree-object.ts

```diff
@@ -8,20 +8,28 @@ export async function valueToTreeObject(
   path: string,
   value: string | File
 ) {
-  let mode = "100644";
-  if (value !== null && typeof value !== "string") {
-    mode = value.mode || mode;
-  }
+  const defaultMode = "100644";
 
   // Text files can be changed through the .content key
   if (typeof value === "string") {
     return {
       path,
-      mode: mode,
+      mode: defaultMode,
       content: value,
     };
   }
 
+  const mode = value.mode ?? defaultMode;
+
+  // UTF-8 files can be treated as text files
+  if (value.encoding === "utf-8") {
```

Comment:
I'd like to add a reference to your pull request as it provides great background information

```suggestion
  // https://github.com/gr2m/octokit-plugin-create-pull-request/pull/133
  if (value.encoding === "utf-8") {
```